lib/iruby/ostream.rb

```diff
@@ -25,26 +25,33 @@ def read(*args)
     alias_method :next, :read
     alias_method :readline, :read
 
-    def write(s)
-      raise 'I/O operation on closed file' unless @session
-      @session.send(:publish, :stream, name: @name, text: s.to_s)
-      nil
+    def write(*obj)
+      str = StringIO.open {|sio| sio.write(*obj); sio.string }
```

Comment:
We can refactor this line by introducing new private method like:

```ruby
def write(*obj)
  str = build_string {|sio| sio.write(*obj) }
end

private

def build_string
  StringIO.open {|sio| yield(sio); sio.string }
end
```

Reply:
Thanks. Done.